pkg/scheduler/framework/plugins/volumebinding/volume_binding.go

```diff
@@ -123,6 +127,41 @@ func (pl *VolumeBinding) EventsToRegister() []framework.ClusterEventWithHint {
 	return events
 }
 
+// TODO: kube-schduler should unregister the CSINode events once all the volume plugins has completed their CSI migration.
+func (pl *VolumeBinding) isSchedulableAfterCSINodeChange(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) (framework.QueueingHint, error) {
+	oldCSINode, modifiedCSINode, err := util.As[*storagev1.CSINode](oldObj, newObj)
+	if err != nil {
+		return framework.Queue, err
+	}
+
+	logger = klog.LoggerWithValues(
+		logger,
+		"Pod", klog.KObj(pod),
+		"CSINode", klog.KObj(modifiedCSINode),
+	)
+
+	oldAnn := oldCSINode.ObjectMeta.Annotations[v1.MigratedPluginsAnnotationKey]
+	modifiedAnn := modifiedCSINode.ObjectMeta.Annotations[v1.MigratedPluginsAnnotationKey]
+
```

Comment:
Can we just compare as string?

```go
if oldCSINode.ObjectMeta.Annotations[v1.MigratedPluginsAnnotationKey] != modifiedCSINode.ObjectMeta.Annotations[v1.MigratedPluginsAnnotationKey] {
  return framework.Queue, nil 
}
```

That could return `Queue` mistakenly when the annotation value is changed from `hoge,fuga` to `fuga,hoge` though, we can say it's a rare case.

Reply:
fix to compare annotations as string.
[4bbb17d](https://github.com/kubernetes/kubernetes/pull/125097/commits/4bbb17da70063aa87554d8986e3b81a06c269867)